### PR TITLE
ADR-1221: View past returns hide table when no previous returns

### PR DESCRIPTION
**Pull request by sini-george**

### CODE REVIEW

File: app/views/returns/ViewPastReturnsView.scala.html [hunk 1]
```diff
@@ -38,42 +39,50 @@
...
-                    text = action.label,
-                    href = action.href,
-                    visuallyHiddenContent = action.visuallyHiddenText
-                    )
-                ))))
-                row.cells :+ actions
-            }
-            ))
-        )
+        @if(outstandingReturns.rows.size > 0) {
```
Comment (davidfes) [inline app/views/returns/ViewPastReturnsView.scala.html:42]: prefer .nonEmpty to size > 0

File: app/viewmodels/returns/ViewPastReturnsHelper.scala [hunk 2]
```diff
@@ -35,11 +35,15 @@ class ViewPastReturnsHelper @Inject() () extends Logging {
...
-      total = None
-    )
+    if (sortedObligationData.nonEmpty) {
+      TableViewModel(
+        head = getTableHeader(messages),
+        rows = getObligationDataTableRows(sortedObligationData),
+        total = None
+      )
+    } else {
+      TableViewModel(head = Seq.empty, Seq.empty, total = None)
```
Comment (davidfes) [inline app/viewmodels/returns/ViewPastReturnsHelper.scala:45]: Wonder if having a constructor method would be better here.

object TableViewModel {
  def empty(): TableViewModel = TableViewModel(Seq.empty, Seq,empty, None)
}

Also a bit inconsistent that the first and third parameter are named, but not the second.

File: app/views/returns/ViewPastReturnsView.scala.html [hunk 3]
```diff
@@ -38,42 +39,50 @@
...
-                    text = action.label,
-                    href = action.href,
-                    visuallyHiddenContent = action.visuallyHiddenText
-                    )
-                ))))
-                row.cells :+ actions
-            }
-            ))
-        )
+        @if(completedReturns.rows.size > 0) {
```
Comment (davidfes) [inline app/views/returns/ViewPastReturnsView.scala.html:63]: same comment here

File: test/viewmodels/checkAnswers/returns/ViewPastReturnsHelperSpec.scala [hunk 4]
```diff
@@ -79,6 +86,17 @@ class ViewPastReturnsHelperSpec extends SpecBase with ScalaCheckPropertyChecks {
       }
     }
 
+    "must return the OverDue status an overdue obligation" in new SetUp {
+      val obligationData = Seq(obligationDataSingleOverDue)
+      val table          = viewPastReturnsHelper.getReturnsTable(obligationData)
+      table.rows.size shouldBe obligationData.size
+      table.rows.map { row =>
```
Comment (davidfes) [inline test/viewmodels/checkAnswers/returns/ViewPastReturnsHelperSpec.scala:93]: since we don't need the result, prefer foreach to map

**Review by davidfes — APPROVED** [review]

File: test/viewmodels/returns/ViewPastReturnsHelperSpec.scala [hunk 5]
```diff
@@ -86,17 +86,6 @@ class ViewPastReturnsHelperSpec extends SpecBase with ScalaCheckPropertyChecks {
       }
     }
 
-    "must return the OverDue status an overdue obligation" in new SetUp {
```
Comment (davidfes) [inline test/viewmodels/returns/ViewPastReturnsHelperSpec.scala]: As noted I've added this test (boundary check) in my unit test jira

**Review by davidfes — APPROVED** [review]

**Review by davidfes — APPROVED** [review]

**Review by msatta — APPROVED** [review]

**Pull request merged**